test/unit/range/views/view_kmer_hash_test.cpp

```diff
@@ -17,155 +17,86 @@
+{
+    TypeParam text{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4}; // ACGTAGC
+    result_t ungapped{6,27,44,50,9};
+    result_t gapped{2,7,8,14,1};
+    EXPECT_EQ(ungapped, text | ungapped_view | seqan3::views::to<result_t>);
+    EXPECT_EQ(gapped, text | gapped_view | seqan3::views::to<result_t>);
+
+    auto stop_at_t = seqan3::views::take_until([] (seqan3::dna4 x) { return x == 'T'_dna4; });
+    EXPECT_EQ(result_t{6}, text | stop_at_t | ungapped_view | seqan3::views::to<result_t>);
+    EXPECT_EQ(result_t{2}, text | stop_at_t | gapped_view | seqan3::views::to<result_t>);
+}
 
-TEST_F(kmer_hash_test, concepts)
+TYPED_TEST(kmer_hash_properties_test, concepts)
 {
-    auto v1 = text1 | ungapped_view;
+    TypeParam text{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4}; // ACGT
+    auto v1 = text | ungapped_view;
     EXPECT_TRUE(std::ranges::input_range<decltype(v1)>);
     EXPECT_TRUE(std::ranges::forward_range<decltype(v1)>);
-    EXPECT_TRUE(std::ranges::bidirectional_range<decltype(v1)>);
-    EXPECT_TRUE(std::ranges::random_access_range<decltype(v1)>);
+    EXPECT_EQ(std::ranges::bidirectional_range<decltype(text)>, std::ranges::bidirectional_range<decltype(v1)>);
+    EXPECT_EQ(std::ranges::random_access_range<decltype(text)>, std::ranges::random_access_range<decltype(v1)>);
     EXPECT_TRUE(std::ranges::view<decltype(v1)>);
-    EXPECT_TRUE(std::ranges::sized_range<decltype(v1)>);
+    EXPECT_EQ(std::ranges::sized_range<decltype(text)>, std::ranges::sized_range<decltype(v1)>);
     EXPECT_FALSE(std::ranges::common_range<decltype(v1)>);
     EXPECT_TRUE(seqan3::const_iterable_range<decltype(v1)>);
     EXPECT_FALSE((std::ranges::output_range<decltype(v1), size_t>));
-
-    auto v2 = text5 | ungapped_view;
-    EXPECT_TRUE(std::ranges::input_range<decltype(v2)>);
-    EXPECT_TRUE(std::ranges::forward_range<decltype(v2)>);
-    EXPECT_TRUE(std::ranges::bidirectional_range<decltype(v2)>);
-    EXPECT_FALSE(std::ranges::random_access_range<decltype(v2)>);
-    EXPECT_TRUE(std::ranges::view<decltype(v2)>);
-    EXPECT_FALSE(std::ranges::sized_range<decltype(v2)>);
-    EXPECT_FALSE(std::ranges::common_range<decltype(v2)>);
-    EXPECT_TRUE(seqan3::const_iterable_range<decltype(v2)>);
-    EXPECT_FALSE((std::ranges::output_range<decltype(v2), size_t>));
-
-    auto v3 = text6 | ungapped_view;
-    EXPECT_TRUE(std::ranges::input_range<decltype(v3)>);
-    EXPECT_TRUE(std::ranges::forward_range<decltype(v3)>);
-    EXPECT_FALSE(std::ranges::bidirectional_range<decltype(v3)>);
-    EXPECT_FALSE(std::ranges::random_access_range<decltype(v3)>);
-    EXPECT_TRUE(std::ranges::view<decltype(v3)>);
-    EXPECT_FALSE(std::ranges::sized_range<decltype(v3)>);
-    EXPECT_FALSE(std::ranges::common_range<decltype(v3)>);
-    EXPECT_TRUE(seqan3::const_iterable_range<decltype(v3)>);
-    EXPECT_FALSE((std::ranges::output_range<decltype(v3), size_t>));
 }
 
 TEST_F(kmer_hash_test, ungapped)
```

Comment:
```suggestion
TYPED_TEST(kmer_hash_ungapped, combined_with_container)
```

Isn't this test already tested in?

```c++
TYPED_TEST(kmer_hash_properties_test, combinability)
//...
    TypeParam text{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4}; // ACGTAGC
    result_t ungapped{6,27,44,50,9};
```

Reply:
@marehr  No, in the typed test is another dna_vector tested, here are the strings "AAAA", "AC" and "ACG" tested.